core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -3694,14 +3693,13 @@ String getVariableName() {
       return variableName;
     }
 
-    @Nullable NullPolicy getNullPolicy() {
+    public NullPolicy getNullPolicy() {
```

Comment:
why add this `public`? IMHO, It should be a similar qualifier to other methods in this static class such as `getCondition`, `getVariableName`. 

and if we remove @Nullable (we expect to pass specific nonnull enumeration values), shall we add a `requireNonNull` checking in constructor? otherwise the below `unboxIfNecessary`method
```
switch (nullPolicy)
```
may cause NPE.

Reply:
Adding the `public` was a mistake. I will revert. (I considered moving `getNullPolicy` up to the interface, but it didn't work.)

Method parameters are implicitly not-nullable, so a `requireNotNull` is not necessary. We sometimes use `requireNonNull` in public APIs (so that users get a nicer exception) but this is not a public API.